### PR TITLE
175378811 Display eruptions

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
Eruptions can be toggled on the map and will display over time in the same way as earthquakes are displayed. There's a lot of code that is mirrored from the Earthquakes implementation, and as such we've probably got some areas that would benefit from combining, but for this round I figured I would stop at the point where things worked as intended and put in for a review before making any further changes. The duplication isn't that harmful I don't think, but I'll wait for review and respond accordingly.

I also looked at the existing volcanoes data and compared to the newer data from Smithsonian - we could potentially replace the embedded data with a dynamic dataset from the server, though there are a couple of fields missing on our server side that are included in the static set in the repository. The semantic difference between the site of a volcano, and the site of eruptions for that named volcano means we can consider and render the two separately (with different information), but for now I have left the older dataset as-is.

### CODE REVIEW

File: js/3d/top-view.js [hunk 1]
```diff
@@ -62,6 +69,10 @@ export default class {
     if (earthquake) {
       return { type: 'earthquake', data: earthquake }
     }
+    const eruption = this.eruptions.spriteAt(x, this._height - y)
+    if (eruption) {
+      return { type: 'eruption', data: eruption }
+    }
```
Comment (pjanik) [inline js/3d/top-view.js:75]: I think it'd be better if that's the first statement in this function. So, an eruption click is detected before volcano.

**Review by pjanik — CHANGES_REQUESTED** [review]
It looks good. Right, lots of copy-paste code, but the division between earthquakes and eruptions is pretty clear.

I've left one small inline comment.

I can also see two issues when I test with this branch:
1. Eruptions are not visible in the cross-section. I guess they should be, right? Volcanoes and earthquakes are.
2. z-ordering of eruptions doesn't seem to be correct, or it's random. When there are multiple overlapping eruptions, the old ones are covering the new ones (for example Etna, Sicily, Italy). Maybe it's just enough to sort the data in the opposite way?

**Comment by dstrawberrygirl** [comment]
Thanks for the review, @pjanik - I have managed to get eruptions to display in cross-section but I can't quite see why the eruptions outside of the 3d cross-section view are not disappearing like the earthquakes are. If you have any hints here I'd be grateful!

File: js/selectors/index.js [hunk 2]
```diff
@@ -84,3 +84,33 @@ export const getVisibleVolcanoes = createSelector(
...
+
+export const getVisibleEruptions = createSelector(
+  [eruptionsEnabled, eruptionsData, filters, crossSectionPoints],
+  (eruptionsEnabled, eruptionsData, filters, crossSectionPoints) => {
+    if (!eruptionsEnabled) {
+      return []
+    }
+    const minTime = filters.get('minTime')
+    const maxTime = filters.get('maxTime')
+    // const crossSectionFilter = getCrossSectionFilter(crossSectionPoints)
```
Comment (pjanik) [inline js/selectors/index.js:98]: I think that's the reason why you see all the eruptions in the cross-section.

**Comment by pjanik** [comment]
> eruptions outside of the 3d cross-section view are not disappearing like the earthquakes are. If you have any hints here I'd be grateful!

I've left an inline comment that should help.

It seems the build is broken because of the lint error, but I guess you've noticed that.

One more thing - did you check if Cypress tests work okay? They're based on screenshots, so it'd be nice to update them when necessary. It could be done in the last PR too.

Thanks!

File: js/selectors/index.js (hunk 2 above)
Comment (dstrawberrygirl) [reply js/selectors/index.js]: Thanks! I had commented that out during testing and never uncommented. Well spotted!

**Comment by dstrawberrygirl** [comment]
Thanks again @pjanik - your suggestion was spot on, and the eruptions should now behave as expected in cross-section view. I have updated some of the Cypress screenshots to pass tests, but I agree a story needs to be added to make sure we have new screenshots when the new UI has all been added. Hopefully this PR will be good to go once Travis is done.

**Review by pjanik — APPROVED** [review]
Looks good, thanks

**Pull request merged**